src/main/java/net/hypixel/nerdbot/listener/PinListener.java

```diff
@@ -48,23 +47,16 @@ public void onMessageReceived(MessageReceivedEvent event) {
             return; // Ignore any message that the thread and message IDs do not match.
         }
 
-        Suggestion suggestion = NerdBotApp.getBot().getSuggestionCache().getSuggestion(event.getChannel().getId());
-
-        if (suggestion == null) {
-            return; // Ignore uncached suggestions
+        // Ignore if feature is globally turned off.
+        ChannelConfig channelConfig = NerdBotApp.getBot().getConfig().getChannelConfig();
+        if (!channelConfig.isAutoPinFirstMessage()) {
+            return;
         }
 
-        // Ignore if feature is globally turned off.
-        if (suggestion.getChannelType() == Suggestion.ChannelType.NORMAL) {
-            SuggestionConfig suggestionConfig = NerdBotApp.getBot().getConfig().getSuggestionConfig();
-            if (!suggestionConfig.isAutoPinFirstMessage()) {
-                return;
-            }
-        } else {
-            AlphaProjectConfig alphaProjectConfig = NerdBotApp.getBot().getConfig().getAlphaProjectConfig();
-            if (!alphaProjectConfig.isAutoPinFirstMessage()) {
-                return;
-            }
+        // Ignore channel if blacklisted.
+        String[] autoPinBlacklistedChannels = channelConfig.getAutoPinBlacklistedChannels();
+        if (Util.safeArrayStream(autoPinBlacklistedChannels).anyMatch(event.getChannel().asThreadChannel().getParentChannel().asForumChannel().getId()::equalsIgnoreCase)) {
```

Comment:
I think the channel ID here should be assigned to a variable so it's easier to read